src/iris/ui/static/js/iris.js

```diff
@@ -2111,6 +2147,61 @@ iris = {
       });
       this.redrawApplicationDropdown();
     },
+    createCustomNotificationTable: function(){
+      var template = Handlebars.compile(this.data.customNotificationTemplate),
+          settings = this.data.settings,
+          self = this;
+      this.data.$customNotificationTable.empty();
+      this.data.$customNotificationTable.append(template(settings));
+      this.data.$customNotificationTable.find('button.delete-custom-app-button').each(function() {
+        $(this).click(function() {
+          self.deletePerCustomApp($(this));
+        });
+      });
+
+      this.data.$customNotificationTable.find('button.edit-custom-app-button').each(function() {
+        $(this).click(function() {
+          self.editPerCustomApp($(this));
+        });
+      });
+
+      this.redrawApplicationNotificationDropdown();
+    },
+    createCustomNotificationModal: function(app){
+
+      var template = Handlebars.compile(this.data.createCustomNotificationModalTemplate),
+              settings = this.data.settings,
+              customNotificationModal = this.data.$customNotificationModal;
+
+      // call out to the back end to get all categories for this app
+      var appNotificationCategories = [];
+      $.ajax({
+        type: "GET",
+        url: '/v0/categories/' + app,
+        success: function(resp)
+        {
+          appNotificationCategories = resp;
+          appNotificationCategories.forEach((appSetting, i) => {
+            settings.categories.forEach(userSetting => {
+              if(appSetting['application'] === userSetting['application'] && appSetting['name'] === userSetting['category']){
+                //apply user overrides
+                appNotificationCategories[i]['mode'] = userSetting['mode'];
+              }
+            });
+
+          });
+          settings['modalAppName'] = app;
+          settings['appCategoryData'] = appNotificationCategories;
+          settings['supported_modes'] = ['email','slack','call','sms','drop'];
```

Comment:
Would it be better if this hard coded list of modes be defined or derived from somewhere else in the codebase? 

With the list being hard coded here, if we ever need to add more modes, we'd need to remember to update this hard coded list. Also, `im` is missing.

Reply:
changed